### PR TITLE
[YPPY-1615] - Add CI actions

### PR DESCRIPTION
**Pull request by FalconerTC**
- Setup github action to automatically release an deploy on merge to master

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,24 @@
...
+    <version>1.16-SNAPSHOT</version>
     <name>Conduit</name>
     <description>
         A simple abstraction over the rabbitmq java client
     </description>
     <url>www.renttherunway.com</url>
 
+    <developers>
+        <developer>
+            <name>Gene Trog</name>
```
Comment (FalconerTC) [inline pom.xml:16]: Return of the king 😛

File: pom.xml [hunk 2]
```diff
@@ -28,28 +39,43 @@
...
                         </executions>
                     </plugin>
+                    <plugin>
+                        <groupId>org.sonatype.plugins</groupId>
+                        <artifactId>nexus-staging-maven-plugin</artifactId>
+                        <version>1.6.8</version>
+                        <extensions>true</extensions>
+                        <inherited>false</inherited>
+                        <configuration>
+                            <serverId>sonatype-nexus-snapshots</serverId> <!-- Change to sonatype-nexus-staging -->
```
Comment (FalconerTC) [inline pom.xml:73]: Publishes to snapshots so I can verify it works. I'll have a follow-up PR to change this.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+          release-branch-name: "master"
+
+          git-release-bot-name: "rtr-actions"
+          git-release-bot-email: "infrastructure+github@renttherunway.com"
+
+          gpg-enabled: "true"
+          gpg-key-id: ${{ secrets.GPG_KEY_ID }}
+          gpg-key: ${{ secrets.GPG_KEY }}
+
+          maven-repo-server-username: ${{ secrets.OSS_NEXUS_USERNAME }}
```
Comment (gtrog) [inline .github/workflows/release.yml:33]: Can you make these org-level?

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      - name: Perform Maven Release
+        uses: qcastel/github-actions-maven-release@master
+        with:
+          release-branch-name: "master"
+
+          git-release-bot-name: "rtr-actions"
+          git-release-bot-email: "infrastructure+github@renttherunway.com"
+
+          gpg-enabled: "true"
+          gpg-key-id: ${{ secrets.GPG_KEY_ID }}
```
Comment (gtrog) [inline .github/workflows/release.yml:30]: Same, can this be org-level, maybe with a more explicit name (`OSS_NEXUS_`...)

File: pom.xml (hunk 1 above)
Comment (gtrog) [reply pom.xml:16]: Why only me!? And I don't know this `Gene Trog` person ;)

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v2
+
+      - name: Install Java (${{ matrix.java }}) and Maven
```
Comment (gtrog) [inline .github/workflows/release.yml:16]: Where is your `strategy` defined that contains `java`?

Comment (FalconerTC) [reply .github/workflows/release.yml:16]: Whoops

File: .github/workflows/release.yml (hunk 3 above)
Comment (FalconerTC) [reply .github/workflows/release.yml:33]: I was going to after I verified the workflow was working correctly as I can't add them myself for testing.

Comment (gtrog) [reply .github/workflows/release.yml:33]: Actually, I just remembered, you can't make them org-level.  You can't have a public repo use org-level secrets for security reasons :/

**Pull request merged**